node/manager_test.go

```diff
@@ -0,0 +1,219 @@
+		tMock.EXPECT().BroadcastOutbound(gomock.Any(), gomock.Any()).Times(0)
+		err := got.Start(context.Background())
+		require.NoError(err)
+		time.Sleep(time.Second)
+	})
+
+	t.Run("enable_broadcast", func(t *testing.T) {
+		cfg := Config{false, 100 * time.Millisecond}
+		got := NewDelegateManager(&cfg, tMock, hMock, pMock)
+		require.Equal(cfg, got.cfg)
+		require.Equal(map[string]Info{}, got.nodeMap)
+		require.Equal(tMock, got.transmitter)
+		require.Equal(hMock, got.heightable)
+		require.Equal(pMock, got.privKey)
+		require.Equal(true, got.broadcaster != nil)
+
+		privK, _ := crypto.GenerateKey()
+		hMock.EXPECT().TipHeight().Return(uint64(10)).AnyTimes()
+		pMock.EXPECT().PublicKey().Return(privK.PublicKey()).AnyTimes()
+		pMock.EXPECT().Sign(gomock.Any()).Return([]byte("xxxxxx"), nil).AnyTimes()
+		tMock.EXPECT().BroadcastOutbound(gomock.Any(), gomock.Any()).Return(nil).MinTimes(1)
+		err := got.Start(context.Background())
+		require.NoError(err)
+		time.Sleep(time.Second)
+	})
+}
+
+func TestDelegateManager_HandleNodeInfo(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+	hMock := mock_node.NewMockheightable(ctrl)
+	tMock := mock_node.NewMocktransmitter(ctrl)
+	pMock := mock_node.NewMockprivateKey(ctrl)
+
+	require := require.New(t)
+	privKey, err := crypto.GenerateKey()
+	require.NoError(err)
+
+	t.Run("verify_pass", func(t *testing.T) {
+		msg := &iotextypes.ResponseNodeInfoMessage{
+			Info: &iotextypes.NodeInfo{
+				Version:   "v1.8.0",
+				Height:    200,
+				Timestamp: timestamppb.Now(),
+				Address:   privKey.PublicKey().Address().String(),
+			},
+		}
+		hash := hashNodeInfo(msg)
+		msg.Signature, _ = privKey.Sign(hash[:])
+		dm := NewDelegateManager(&DefaultConfig, tMock, hMock, pMock)
+		dm.HandleNodeInfo(context.Background(), "abc", msg)
+		addr := msg.Info.Address
+		require.Equal(msg.Info.Height, dm.nodeMap[addr].Height)
+		require.Equal(msg.Info.Version, dm.nodeMap[addr].Version)
+		require.Equal(msg.Info.Timestamp.AsTime().String(), dm.nodeMap[addr].Timestamp.String())
+		m := dto.Metric{}
+		nodeDelegateHeightGauge.WithLabelValues(addr, msg.Info.Version).Write(&m)
+		require.Equal(msg.Info.Height, uint64(m.Gauge.GetValue()))
+	})
+
+	t.Run("verify_unpass", func(t *testing.T) {
+		privKey2, _ := crypto.GenerateKey()
+		msg := &iotextypes.ResponseNodeInfoMessage{
+			Info: &iotextypes.NodeInfo{
+				Version:   "v1.8.0",
+				Height:    200,
+				Timestamp: timestamppb.Now(),
+				Address:   privKey2.PublicKey().Address().String(),
+			},
+		}
+		hash := hashNodeInfo(msg)
+		msg.Signature, _ = privKey.Sign(hash[:])
+		dm := NewDelegateManager(&DefaultConfig, tMock, hMock, pMock)
+		dm.HandleNodeInfo(context.Background(), "abc", msg)
+		addr := msg.Info.Address
+
+		_, ok := dm.nodeMap[addr]
+		require.False(ok)
+		m := dto.Metric{}
+		nodeDelegateHeightGauge.WithLabelValues(addr, msg.Info.Version).Write(&m)
+		require.Equal(uint64(0), uint64(m.Gauge.GetValue()))
```

Comment:
this is to test wrong name, can you add another testcase where the signature failed verification? you can simply generate a hard-code sig value and use it, since every time the test is generating a new key, it will fail the sig verification